mayastor/src/bdev/nexus/nexus_bdev_children.rs

```diff
@@ -476,13 +475,9 @@ impl Nexus {
             });
         }
 
-        // FIXME: use dummy key for now
         for c in &self.children {
-            if let Err(e) = c.resv_register(0x12345678).await {
-                error!(
-                    "{}: child {} failed to register key {}",
-                    self.name, c.name, e
-                );
+            if let Err(e) = c.acquire_write_exclusive().await {
+                error!("Failed to acquire write exclusive reservation on child: {}", e);
```

Comment:
What is our fallback scenario here if we fail to acquire exclusive access ? Do we keep using such a child ? Won't this lead to data corruption if the child is being used by other nexus ?

Reply:
Currently we do blunder ahead and attempt to use it but if the write exclusive reservation is held by another host then all writes will fail and it'll be unusable. I'll change it to return an error.

Reply:
I'm going to refactor this slightly. Currently `NexusChild::acquire_write_exclusive()` is called from `Nexus::try_open_children()` and `Nexus::add_child_only()`, and I think it's better to call it from `NexusChild::open()` directly to cover both cases so if the write exclusive reservation fails, it'll fail as part of opening the child and allow the existing code to handle clean-ups. This does also mean it'll try to acquire the write exclusive reservation in `NexusChild::online()` but I think that should be fine and onlining a child is only possible via gRPC (which in turn that may be dead code).